src/validate_async.cc

```diff
@@ -0,0 +1,113 @@
+#include <string>
+#include <sstream>
+#include "v8_wrapper.h"
+#include "protagonist.h"
+#include "drafter.h"
+#include "RefractDataStructure.h"
+#include "ConversionContext.h"
+
+using std::string;
+using namespace v8;
+using namespace protagonist;
+
+// Async Validate
+void AsyncValidate(uv_work_t* request);
+
+// Async Validate Handler
+void AsyncValidateAfter(uv_work_t* request);
+
+// Threadpooling libuv baton
+struct Baton {
+
+    // Callback
+    Nan::Persistent<Function> callback;
+
+    // Input
+    mdp::ByteBuffer sourceData;
+
+    // Output
+    sos::Object result;
+};
+
+NAN_METHOD(protagonist::Validate) {
+    Nan::HandleScope scope;
+
+    // Check arguments
+    if (info.Length() != 1 && info.Length() != 2) {
+        Nan::ThrowTypeError("wrong number of arguments, `validate(string, callback)` expected");
+        return;
+    }
+
+    if (!info[0]->IsString()) {
+        Nan::ThrowTypeError("wrong argument - string expected, `validate(string, callback)`");
+        return;
+    }
+
+    if (info.Length() == 2 && !info[1]->IsFunction()) {
+        Nan::ThrowTypeError("wrong argument - callback expected, `validate(string, callback)`");
+        return;
+    }
+
+    // Get source data
+    String::Utf8Value sourceData(info[0]->ToString());
+
+    // Get Callback
+    Local<Function> callback = Local<Function>::Cast(info[1]);
+
+    // Prepare threadpool baton
+    Baton* baton = ::new Baton();
+    baton->sourceData = *sourceData;
+    baton->callback.Reset(callback);
+
+    // This creates the work request struct.
+    uv_work_t *request = ::new uv_work_t();
+    request->data = baton;
+
+    // Schedule the work request
+    int status = uv_queue_work(uv_default_loop(),
+                                request,
+                                AsyncValidate,
+                                (uv_after_work_cb)AsyncValidateAfter);
+
+    assert(status == 0);
```

Comment:
are you aware that the assert is removed from the release build and so there is effectively no check for status in release version?